crates/services/txpool_v2/src/service.rs

```diff
@@ -451,6 +456,45 @@ where
         }
     }
 
+    fn process_tx_update(&self, tx_id: TxId, tx_status: TransactionStatus) {
```

Comment:
I think we just should work with `PreconfirmationStatus` for simplicity=)

Reply:
Done with a special status to be able to not change the Arc type that is in tx status manager and avoid a clone